resource_transip_private_network.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/transip/gotransip/v6"
+	"github.com/transip/gotransip/v6/repository"
+	"github.com/transip/gotransip/v6/vps"
+)
+
+func resourcePrivateNetwork() *schema.Resource {
+	return &schema.Resource{
+		Create: resourcePrivateNetworkCreate,
+		Read:   resourcePrivateNetworkRead,
+		// Update: resourcePrivateNetworkUpdate,
+		Delete: resourcePrivateNetworkDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Required: true,
+				Type:     schema.TypeString,
+				ForceNew: true,
+			},
+			"name": {
+				Computed: true,
+				Type:     schema.TypeString,
+			},
+			"is_blocked": {
+				Computed: true,
+				Type:     schema.TypeString,
+			},
+			"is_locked": {
+				Computed: true,
+				Type:     schema.TypeString,
+			},
+			"vps_names": {
+				Computed: true,
+				Type:     schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourcePrivateNetworkCreate(d *schema.ResourceData, m interface{}) error {
+	description := d.Get("description").(string)
+	client := m.(repository.Client)
+	repository := vps.PrivateNetworkRepository{Client: client}
+
+	err := repository.Order(description)
+	if err != nil {
+		return fmt.Errorf("failed to order private network %s: %s", description, err)
+
+	}
+	setPrivateNetworkID(d, m)
```

Comment:
Can you inline this code for now. I'm still trying to figure out what parts can be put into reusable functions. But for now I'd like all code in the function. 

Also you are ignoring the return value of `setPrivateNetworkID` which will break the functionality of the retry in case of a non-happy flow.